browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,236 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/split_view/split_view_location_bar.h"
+
+#include <algorithm>
+#include <memory>
+#include <utility>
+
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/ui/color/brave_color_id.h"
+#include "brave/components/vector_icons/vector_icons.h"
+#include "cc/paint/paint_flags.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/base/models/image_model.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/geometry/point.h"
+#include "ui/views/background.h"
+#include "ui/views/border.h"
+#include "ui/views/controls/image_view.h"
+#include "ui/views/controls/label.h"
+#include "ui/views/layout/box_layout_view.h"
+#include "ui/views/layout/fill_layout.h"
+#include "ui/views/widget/widget_delegate.h"
+#include "url/url_constants.h"
+
+SplitViewLocationBar::SplitViewLocationBar() {
+  views::Builder<SplitViewLocationBar>(this)
+      .SetBorder(
+          views::CreateEmptyBorder(gfx::Insets().set_bottom(5).set_right(5)))
+      .SetLayoutManager(std::make_unique<views::FillLayout>())
+      .AddChild(
+          views::Builder<views::BoxLayoutView>()
+              .SetBetweenChildSpacing(8)
+              .SetBorder(views::CreateEmptyBorder(
+                  gfx::Insets().set_top(1).set_bottom(4).set_left_right(12, 8)))
+              .AddChild(views::Builder<views::ImageView>()
+                            .CopyAddressTo(&safety_icon_)
+                            .SetImage(ui::ImageModel::FromVectorIcon(
+                                kLeoWarningTriangleOutlineIcon,
```

Comment:
~~I wonder if its worth using the (overridden) Chromium icon here instead, so this always stays in sync with the Omnibox~~ nvm - I checked and it uses the filled icon so we can't reuse it.

:+1: